components/studio/bin/hab-studio.ps1

```diff
@@ -298,8 +298,96 @@ function Enter-Studio {
     $env:HAB_GLYPH_STYLE="ascii"
   }
   New-Studio
-  Write-HabInfo "Entering Studio at $HAB_STUDIO_ROOT"
   $env:STUDIO_SCRIPT_ROOT = $PSScriptRoot
+  $shouldStartStudio = $false
+
+  if(!(Test-InContainer) -and (Get-Process -Name hab-sup -ErrorAction SilentlyContinue)) {
+    Write-Warning "A Habitat Supervisor is already running on this machine."
+    Write-Warning "Only one Supervisor can run at a time."
+    Write-Warning "A Supervisor will not be started in this Studio."
+  } else {
+    $habSvc = Get-Service Habitat -ErrorAction SilentlyContinue
+    if(!$habSvc -or ($habSvc.Status -eq "Stopped")) {
+      $shouldStartStudio = $true
+
+      # Set console encoding to UTF-8 so that any redirected glyphs
+      # from the supervisor log are propperly encoded
+      [System.Console]::OutputEncoding = [System.Text.Encoding]::UTF8
+
+      # We start the Supervisor and handle its output logging in C# which will handle
+      # the process's OutputDataReceived and ErrorDataReceived events in a separate thread.
+      # While we could use PowerShell's Register-ObjectEvent instead. That uses a PSJob
+      # which will be blocked while the interactive studio is open
+      $c = @"
+      using System;
```

Comment:
I'm generally opposed to embedding other languages inside shell strings like this unless it's absolutely necessary. Could this be a standalone C# file? I don't see any interpolation, but I'm pretty ignorant with respect to pwsh.

Reply:
fair point. Moved to its own file @baumanj